EmcraftSystems/linux-stm
# RM 6776: Increase stm32 sdmmc clock max freq

### Design

The max freq was changed for the `stm32_sdmmcv2` variant (variant_stm32_sdmmcv2) in the ARM AMBA MMC Interface driver:

```
commit b2a9fa484fa5393d21fe67406af8a9ab7616bf42
Author: Yann Gautier <yann.gautier@foss.st.com>
Date:   Thu Mar 25 15:05:31 2021 +0100

    mmc: mmci: increase stm32 sdmmcv2 clock max freq

    The variant->f_max is dependent on the IP, not on the SoC where it is
    embedded. Set the max frequency of its source clock to 267MHz.
    The frequency used will be limited by the IOs max frequency, set in the
    SoC device tree.

    Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
    Change-Id: Idc77e9e0a40269ded33e7e2d4536ff4af55e9632
    Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
    Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/228364
```

We will propagate this change to the `stm32_sdmmc` variant (variant_stm32_sdmmc) as well.

Without this change, the driver attempts to change the rate of the SDMMC clock source (to no purpose), which results in the source clock changing from pll1_q_ck to pll2_r_ck. Sometimes this change causes the SD card to hang after reset.

### Test

1. Configure the U-Boot to boot the kernel image automatically from the SD card, and configure `/etc/rc` to reboot in 5 seconds after the system starts.
2. Reset the system so that it enters the boot loop.
3. If the SD card fails then the booting will stop in U-Boot (which should not happen).
4. Run the test for at least 10 hours and make sure that boot loop doesn't stop.
